controllers/release/adapter.go

```diff
@@ -751,20 +912,32 @@ func (a *adapter) validatePipelineSource() *controller.ValidationResult {
 		return a.validationError(err)
 	}
 
-	if releasePlan.Spec.Pipeline != nil {
-		pipelineRef = releasePlan.Spec.Pipeline.PipelineRef
-	} else {
+	if releasePlan.Spec.TenantPipeline != nil {
+		pipelineRef = releasePlan.Spec.TenantPipeline.PipelineRef
+		if !a.releaseServiceConfig.Spec.Debug && pipelineRef.IsClusterScoped() {
+			a.release.MarkValidationFailed("tried using debug only options while debug mode is disabled in the ReleaseServiceConfig")
+			return &controller.ValidationResult{Valid: false}
+		}
+	}
+
+	if releasePlan.Spec.FinalPipeline != nil {
+		pipelineRef = releasePlan.Spec.FinalPipeline.PipelineRef
+		if !a.releaseServiceConfig.Spec.Debug && pipelineRef.IsClusterScoped() {
+			a.release.MarkValidationFailed("tried using debug only options while debug mode is disabled in the ReleaseServiceConfig")
+			return &controller.ValidationResult{Valid: false}
+		}
+	}
+
+	if releasePlan.Spec.Target != "" {
 		releasePlanAdmission, err := a.loader.GetActiveReleasePlanAdmissionFromRelease(a.ctx, a.client, a.release)
 		if err != nil {
 			return a.validationError(err)
 		}
-
 		pipelineRef = releasePlanAdmission.Spec.Pipeline.PipelineRef
```

Comment:
I think you still need a if pipeline != nil check before this line. I am now wondering what would happen if a ReleasePlan has a target and a matching RPA, but the RPA has no pipeline in it. I think that should fail somewhere, but no idea if we catch it anywhere. Probably not in scope of this PR though

Reply:
Ok, I have added in the if condition.

Reply:
I am curious about what you say above going to test it out

Reply:
I created a ticket for it here https://issues.redhat.com/browse/RELEASE-1212